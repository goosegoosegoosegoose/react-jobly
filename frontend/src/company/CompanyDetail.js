import React, {useState, useEffect} from "react";
import { useParams } from "react-router-dom";
import JoblyApi from "../api"
import JobCard from "../jobs/JobCard";

const CompanyDetail = ({apply}) => {
  const { handle } = useParams();
  const [company, setCompany] = useState(null);


  useEffect(() => {
    const fetchCompany = async () => {
      const res = await JoblyApi.getCompany(handle);
      setCompany(res);
    }
    fetchCompany();
  }, []);

  if (!company) {
    return (
      <p>Data is loading...</p>
    );
  }

  return (
    <div>
      <h4>{company.name}</h4>
      <p>{company.description}</p>
        {company.jobs.map((job) => {
          return <JobCard 
            key={job.id}
            id={job.id} 
            title={job.title} 
            salary={job.salary} 
            equity={job.equity}
            apply={apply}
          />})}
    </div>
  )
}

export default CompanyDetail;
