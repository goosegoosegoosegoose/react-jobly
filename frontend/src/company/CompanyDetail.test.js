import { render } from '@testing-library/react';
import CompanyDetail from './CompanyDetail';
import { MemoryRouter } from 'react-router-dom';

test('smoke test', () => {
  render(
    <MemoryRouter>
      <CompanyDetail />
    </MemoryRouter>
  );
});

test('snapshot', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <CompanyDetail />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});