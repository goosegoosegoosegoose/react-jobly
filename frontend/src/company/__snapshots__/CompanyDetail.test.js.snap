// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <p>
    Data is loading...
  </p>
</DocumentFragment>
`;
